src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -272,7 +272,12 @@ private static ClassifiedSpan[] ConvertMultiLineToSingleLineSpans(SourceText tex
                     }
 
                     var updatedClassifiedSpan = new ClassifiedSpan(textSpan.Value, classificationType);
-                    updatedClassifiedSpans.Add(updatedClassifiedSpan);
+
+                    // Omit 0-length spans created in this fashion.
+                    if (updatedClassifiedSpan.TextSpan.Length != 0)
```

Comment:
Can we avoid creating the classified span by checking `textSpan.Value.Length` above?